dashboard/app/controllers/pd/workshop_daily_survey_controller.rb

```diff
@@ -86,6 +66,61 @@ def new_general
       end
     end
 
+    # General workshop daily survey using foorm system.
+    # GET '/pd/workshop_survey/foorm/day/:day'
+    # Where day 0 is the pre-workshop survey, and days 1-5 are the 1st through 5th sessions (index 0-4)
+    #
+    # Currently only generates a day 0 survey, any other day will redirect to a 404.
+    # If survey has been completed already will redirect to thanks page.
+    def new_general_foorm
+      workshop = get_workshop_for_new_general(params[:enrollmentCode], current_user)
+      day = params[:day].to_i
+      if day > 0
+        return render_404
+      end
+
+      unless validate_new_general_parameters(workshop)
+        return
+      end
+
+      session = get_session_for_workshop_and_day(workshop, day)
+
+      # once we have surveys per day parameterize this on day number
+      survey_name = "surveys/pd/workshop_daily_survey_day_0"
+      key_params = {
+        environment: Rails.env,
+        userId: current_user.id,
+        workshopId: workshop.id,
+        day: day,
+        sessionId: session&.id,
+      }
+
+      if Pd::WorkshopSurveyFoormSubmission.has_submitted_form?(current_user.id, workshop.id, session&.id, day, survey_name)
+        return redirect_general(key_params)
+      end
+
+      form, latest_version = Foorm::Form.get_form_and_latest_version_for_name(survey_name)
+      form_data = JSON.parse(form.questions)
+
+      @script_data = {
+        props: {
+          formData: form_data,
```

Comment:
It might be good to settle on "form data" vs. "form questions" as that information moves through our system.

Reply:
true, I think form questions is probably the most clear, what do you think?

Reply:
I like that too.